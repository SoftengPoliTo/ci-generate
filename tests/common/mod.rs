use std::{path::Path, fs};
use walkdir::WalkDir;

pub fn compare_template_output_with_expected_one(snapshot_path: &str, template_path: &str) {

    for entry in WalkDir::new(template_path)
    .follow_links(true)
    .into_iter()
    .filter_map(|e| e.ok()) {
        if entry.path().is_file() {
            let content = fs::read_to_string(Path::new(entry.path())).unwrap();
            let name = entry.path().file_name().unwrap().to_str().unwrap();

            insta::with_settings!({
                snapshot_path => Path::new(snapshot_path)
                .join(entry.path().strip_prefix(Path::new(template_path)).unwrap())
                .parent()
                .unwrap(),
                omit_expression => true
            },{
                insta::assert_snapshot!(name, content)
            });
        }
    }
}
