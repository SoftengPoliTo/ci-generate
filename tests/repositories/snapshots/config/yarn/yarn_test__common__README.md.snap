---
source: tests/common/mod.rs
---
# Personal_Project

[![Actions Status][actions badge]][actions]
[![CodeCov][codecov badge]][codecov]
[![LICENSE][license badge]][license]

Personal_Project's description

# Acknowledgements

<!-- Links -->
[actions]: https://github.com/<your-account>/Personal_Project/actions
[codecov]: https://codecov.io/gh/<your-account>/Personal_Project
[license]: LICENSES/EUPL-1.2.txt

<!-- Badges -->
[actions badge]: https://github.com/<your-account>/Personal_Project/workflows/Personal_Project/badge.svg
[codecov badge]: https://codecov.io/gh/<your-account>/Personal_Project/branch/master/graph/badge.svg
[license badge]: https://img.shields.io/badge/license-EUPL-1.2-blue.svg
